serverlessworkflow/modules/ROOT/pages/eventing/event-correlation-with-workflows.adoc

```diff
@@ -1 +1,216 @@
-//= Event correlation with workflows
\ No newline at end of file
+= Event correlation with workflows
+=====================================
+:compat-mode!:
+// Metadata:
+:description: Event Correlation in Serverless Workflow
+:keywords: kogito, workflow, serverless, correlation, association, domain, events
+
+[[ref-example-jq-expression-switch-conditions]]
+
+Event correlation plays a significant role in large event-driven applications. It allows matching one or more events with a particular workflow instance. Correlation rules with domain information can be used when defining consumed events to match a workflow instance. It is a more practical alternative to the internal identifier `processInstanceId` since you can use any external domain identifier as a correlation.
+
+A correlation definition consists of one or more attributes associated with an event and the respective workflow to which it belongs.  As described in the link:{spec_doc_url}#correlation-definition[Serverless Workflow Specification], the correlation property defines the possible correlations for a given event. Each element should contain a `contextAttributeName` set for the value that should match an attribute from an event.
+
+You can optionally set a `contextAttributeValue` property that should match the value for the respective attribute defined in `contextAttributeName` for the consumed events in a workflow.
+
+[NOTE]
+====
+The incoming events consumed by the workflow engine should contain the correlation attributes, set in the definition, as extension context attributes, in compliance with the link:{cloud_events_url}[Cloud Event] format, thus they are not part of the event payload.
+====
+
+When starting a new workflow instance, the correlation attributes set in the event definition, are extracted from the event and associated with the workflow instance which is being created. Thus, the start event does not trigger a correlation evaluation, instead, it is where correlation attributes and values are set, which are used to evaluate against other incoming events that might trigger this instance.
```

Comment:
I miss the information `who` starts this new workflow instance (as it is basically an engine itself). I understand it in a way that first an event is received by the service, then each workflow with its event and correlation definitions is tried to be matched against the received event and if there is a match, a new instance of the matching workflow is started and the correlation values are set. If there is an already running instance of the matched workflow, it continues the execution.

Reply:
The workflow is started once an event that is associated with the Event state set as a start in the workflow is received. So the who is the one publishing the event is similar to the one who might send a POST request to start a new workflow instance via REST...

Reply:
Yes, my point was to maybe write it in the document so it is more obvious.